stl/inc/xmemory

```diff
@@ -1554,6 +1554,55 @@ namespace ranges {
         && _No_throw_forward_iterator<iterator_t<_Rng>>;
     // clang-format on
 
+    template <class _InIt, class _OutIt>
+    in_out_result<_InIt, _OutIt> _Copy_memcpy_count(_InIt _IFirst, _OutIt _OFirst, const size_t _Count) noexcept {
+        const auto _IFirstPtr     = _To_address(_IFirst);
+        const auto _OFirstPtr     = _To_address(_OFirst);
+        const auto _IFirst_ch     = const_cast<char*>(reinterpret_cast<const volatile char*>(_IFirstPtr));
+        const auto _OFirst_ch     = const_cast<char*>(reinterpret_cast<const volatile char*>(_OFirstPtr));
+        const size_t _Count_bytes = _Count * sizeof(iter_value_t<_InIt>);
+        _CSTD memcpy(_OFirst_ch, _IFirst_ch, _Count_bytes);
+        if constexpr (is_pointer_v<_InIt>) {
+            _IFirst = reinterpret_cast<_InIt>(_IFirst_ch + _Count_bytes);
+        } else {
+            _IFirst += _Count;
+        }

```

Comment:
Does the `is_pointer_v` brach have any runtime benefit over the else branch.

I kind of doubt it and would suggest to simplify to `_IFirst += _Count;`